### PR TITLE
Updates to access control documentation

### PR DESCRIPTION
**Pull request by dcousens**

### CODE REVIEW

File: docs/pages/docs/apis/access-control.mdx [hunk 1]
```diff
@@ -120,20 +125,21 @@ export default config({
...
 
-?> Filter based access control can impact the performance when querying related items on a list.
-Keystone optimises queries for to-one relationships, however these optimisations cannot be used in conjunction with access control filters.
+?> Filter based access control can impact the performance of your database queries.
 
 ### Item (mutations only)
 
-`item` is the final and most powerful level of access control.
-It’s available to `create`, `update`, and `delete` mutations, and lets you write functions which have access to:
+Item-level access control lets you control which mutative operations can be applied to a list item.
```
Comment (flexdinesh) [inline docs/pages/docs/apis/access-control.mdx:135]: > mutative

Would _mutable_ be more apt here?

**Review by flexdinesh — APPROVED** [review]

File: docs/pages/docs/apis/access-control.mdx (hunk 1 above)
Comment (borisno2) [reply docs/pages/docs/apis/access-control.mdx:135]: Or just
```
Item-level access control lets you control which operations can be applied to a list item
```
OR
```
Item-level access control lets you control which mutations can be applied to a list item
```
My preference is 1...

File: docs/pages/docs/apis/access-control.mdx [hunk 2]
```diff
@@ -120,20 +125,21 @@ export default config({
...
+Individual functions can be provided for each of the operations, with each function receiving:
 
-- the **input data** of the mutation (for `create` and `update` operations), and
+- the **input data** of the mutation (for `create` and `update` operations), and or
```
Comment (borisno2) [inline docs/pages/docs/apis/access-control.mdx:138]:
```suggestion
- the **input data** of the mutation (for `create` and `update` operations), and/or
```

File: docs/pages/docs/apis/access-control.mdx [hunk 3]
```diff
@@ -25,17 +25,18 @@ export default config({
...
 
-List-level access control can be specified in three ways.
+There are three types of access control that can be configured on the `access` object:
 
-- [`operation`](#operation) access control lets you check the information in the `context` and `session` objects to decide if the
-  user is allowed to access the list.
-- [`filter`](#filter) access control lets you provide a [GraphQL filter](./filters) to restrict the items the user is allowed to access.
-- [`item`](#item-mutations-only) access control lets you write a function which inspects the provided input data and the existing object (if it exists)
-  and make a decision based on this extra data.
+- [`operation`](#operation), functions which provide the `context` and `session` to help decide if a request should be granted access to the list; this happens prior to any GraphQL query being executed.
```
Comment (borisno2) [inline docs/pages/docs/apis/access-control.mdx:33]:
```suggestion
- [`operation`](#operation), functions that provide the `context` and `session` to help decide if a request should be granted access to the list; this happens prior to any GraphQL query being executed.
```

File: docs/pages/docs/apis/access-control.mdx [hunk 4]
```diff
@@ -25,17 +25,18 @@ export default config({
...
+- [`filter`](#filter), functions which return a [filter](./filters) that is applied to database queries on this list.
+- [`item`](#item-mutations-only), functions which provides the `context` and `session` to determine if a request can execute the respective mutative action.
```
Comment (borisno2) [inline docs/pages/docs/apis/access-control.mdx:35]:
```suggestion
- [`item`](#item-mutations-only), functions that provide the `context` and `session` to determine if a request can execute the respective mutative action.
```

**Review by borisno2 — APPROVED** [review]

**Pull request merged**